modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -810,7 +816,7 @@ private String _getSortableFieldName(String name) {
 	private String _getSortableValue(
 		DDMFormField ddmFormField, Locale locale, Serializable value) {
 
-		if (Validator.isNull(value)) {
+		if (value == null) {
 			return null;
 		}
```

Comment:
@noornajj @ericyanLr based on your comment:

>This occurs because repeatable fields are returned as arrays and when the field is empty, an empty string is added to the array --> [""]
> _getSortableValue will return the empty string as null and then null is added to the string bundler as a literal.
> 
> To fix the issue, I had _getSortableValue return an empty string instead of null.

this change would be enough to fix the bug. Why did you need the other changes?

Reply:
The other changes were to account for extra spaces that were added to the string bundler when the value was an empty string